WordPress/src/test/java/org/wordpress/android/ui/sitecreation/MainVM/NewSiteCreationMainVMTest.kt

```diff
@@ -0,0 +1,209 @@
+
+    @Before
+    fun setUp() {
+        whenever(stepsProvider.getSteps()).thenReturn(SITE_CREATION_STEPS)
+        viewModel = NewSiteCreationMainVM(stepsProvider, tracker)
+        viewModel.start(null)
+        viewModel.navigationTargetObservable.observeForever(navigationTargetObserver)
+        viewModel.wizardFinishedObservable.observeForever(wizardFinishedObserver)
+    }
+
+    @Test
+    fun skipClickedResultsInNextStep() {
+        viewModel.onSkipClicked()
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun segmentSelectedResultsInNextStep() {
+        viewModel.onSegmentSelected(SEGMENT_ID)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun verticalSelectedResultsInNextStep() {
+        viewModel.onVerticalsScreenFinished(VERTICAL_ID)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun siteInfoFinishedResultsInNextStep() {
+        viewModel.onInfoScreenFinished(SITE_TITLE, null)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun domainSelectedResultsInNextStep() {
+        viewModel.onDomainsScreenFinished(DOMAIN)
+        verifyNavigatedToNextStep()
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedSegment() {
+        viewModel.onSegmentSelected(SEGMENT_ID)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.segmentId).isEqualTo(SEGMENT_ID)
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedVertical() {
+        viewModel.onVerticalsScreenFinished(VERTICAL_ID)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.verticalId).isEqualTo(VERTICAL_ID)
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSiteInfo() {
+        viewModel.onInfoScreenFinished(SITE_TITLE, SITE_TAGLINE)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.siteTitle).isEqualTo(SITE_TITLE)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.siteTagLine).isEqualTo(SITE_TAGLINE)
+    }
+
+    @Test
+    fun siteCreationStateUpdatedWithSelectedDomain() {
+        viewModel.onDomainsScreenFinished(DOMAIN)
+        assertThat(viewModel.navigationTargetObservable.lastEvent!!.wizardState.domain).isEqualTo(DOMAIN)
+    }
+
+    @Test
+    fun wizardFinishedInvokedOnSitePreviewCompleted() {
+        val state = SiteCreationCompleted(LOCAL_SITE_ID)
+        viewModel.onSitePreviewScreenFinished(state)
+
+        val captor = ArgumentCaptor.forClass(CreateSiteState::class.java)
+        verify(wizardFinishedObserver).onChanged(captor.capture())
+
+        assertThat(captor.value).isEqualTo(state)
+    }
+
+    @Test
+    fun nextStepAfterOnBackPressedIsCurrentStep() {
+        val startingStep = currentStep()
+        viewModel.onBackPressed() // navigate to previous step
+        viewModel.onSkipClicked() // navigate to next step
+        assertThat(currentStep()).isEqualTo(startingStep)
+    }
+
+    @Test
+    fun backSupressedOnlyForLastStep() {
```

Comment:
```suggestion
    fun backSuppressedOnlyForLastStep() {
```

Nitpick: There is a spelling error and AS is bugging me about it :(

Reply:
Fixed in [fdf52bb](https://github.com/wordpress-mobile/WordPress-Android/pull/9142/commits/fdf52bb7af77f0e58431f87f3616f4bc9ed465c7)